RentACarWebAPI/Services/ClientService.cs

```diff
@@ -1,13 +1,49 @@
 ﻿using RentACarWebAPI.Interfaces.Repositories;
 using RentACarWebAPI.Interfaces.Services;
 using RentACarWebAPI.Models;
+using System.Linq;
 
 namespace RentACarWebAPI.Services
 {
     public class ClientService : BaseService<Client>, IClientService
     {
+        private readonly IClientRepository _clientRepository;
+
         public ClientService(IClientRepository clientRepository) : base(clientRepository)
         {
+            _clientRepository = clientRepository;
+        }
+
+        public override Client Create(Client newClient)
+        {
+            var clientList = _clientRepository.GetAll();
+
+            if (clientList.Any(e => e.Dni == newClient.Dni))
+                return null;
+
+            var clientCreated = base.Create(newClient);
+
+            return clientCreated;
+        }
+
+        public override Client Update(Client newClient, int id)
+        {
+            var existingClient = _clientRepository.Get(id);
+
+            if (existingClient is null)
+                return null;
+
+            var clientList = _clientRepository.GetAll();
```

Comment:
In a real case, maybe with a bigger db of clients, Is it good to use the `Get(id)` method and then use `GetAll()` method?
Or should i `GetAll()` clients and then iterate there?

Reply:
you never want to load an entire table in memory just to perform a check (there may be some scenario but it's an edge one). whatever you plan to check in memory you should be able to do it at the database (making several queries is always an option)

Reply:
correct, you would query the database to see if a record exists with the same DNI and act accordingly